### ISSUE TITLE
Atom feed

### ISSUE THREAD

**Issue by tilsammans** [codevise/pageflow#585]
What would you think about an atom feed?

`GET /entries.atom` would contain the 20 latest published revisions/entries.

**Comment by tf** [comment]
Yes, this would be great. We have also bounced around ideas regarding this feature. Some thoughts:

* We should integrate with `Pageflow.confg.public_entry_request_scope` to allow scoping entries by account etc. (i.e. `my-account.some-pageflow.com/entries.atom`).
* Maybe there could even be a specific or additional `Pageflow.config.entries_feed_request_scope`. That would allow to further restrict which entries show up in the feed without having to anticipate every possible use case in advance.
* Ideally the feed would support some form of pagination instead of truncating the list. But that would surely not be MVP.

**Comment by tilsammans** [comment]
Cool. I know about the `public_entry_request_scope` indeed, that would be my plan of attack as well.

Not sure about the feed scope, but maybe once we use it it will become desirable.

The pagination could be part of the initial approach I think, it shouldn't be too difficult.

I can make time to work on this in the coming weeks.

**Comment by tf** [comment]
That's great to hear! Regarding feed scope: At least for Hosted Pageflow we have some customers that mainly use password protected entries for internal usage. Even if it is just security by obscurity, I could imaging those customers not wanting to see a publicly visible list of entries. Given something like feed scope, we would be able to filter password protected entries or add an `enable_feed` attribute to `Account` via `Pageflow.config.admin_form_input` without having to change any of the logic inside the gem.

**Comment by tilsammans** [comment]
WIP at https://github.com/scrollytelling/pageflow/tree/atom-feed

Questions that come up:

* what to render in the `content` field? all text from all pages, and all photographs and all videos?
* we don't require a pagination library at the moment. ActiveAdmin depends on kaminari. ok to start depending on that as well?

**Comment by tf** [comment]
Hey Joost,

sorry for the delay. Was on vacation for a week. 

* Maybe we can use the [social share description](https://github.com/codevise/pageflow/blob/master/app/helpers/pageflow/social_share_helper.rb#L46) as content?
* Depending on Kaminari seems fine with me. Not aware of a better alternative and it's already in our dependencies.
* We should filter out entries that have been published with a password protection. I've implemented the corresponding [scopes in a concern](https://gist.github.com/tf/3024227ab358793794282cf52272664a) for Hosted Pageflow that we could easily turn into a PR.

**Comment by tilsammans** [comment]
Thinking a little more about this.

If I'm honest, restricting the entries to only those in the theming account is the only sane **account-focused** option here. I am specifically wearing my account manager hat; since we want to sell this feature to her, and she pays our bills. So I've hardcoded it in the controller for now.

A second type of feed might be an all-encompassing one that contains all published entries. Nice to have, especially for your and my business. But having the option that some other customer's stories might end up in my account's Atom feed is unthinkable! People *will* make this mistake.

I feel that my work is nearing completion and that the hard decisions have been made.
